File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Slot.py
from typing import Optional, List
from pydantic import *
from ..base.CodeableConcept import CodeableConcept
from ..base.DomainResource import DomainResource
from ..base.Reference import Reference
from ..base.Identifier import Identifier

class Slot(DomainResource):
    schedule: Reference
    service_category: Optional[List[CodeableConcept]] = None
    specialty: Optional[List[CodeableConcept]] = None
    start: str
    service_type: Optional[List[CodeableConcept]] = None
    appointment_type: Optional[CodeableConcept] = None
    status: str
    comment: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    end: str
    overbooked: Optional[bool] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/Slot.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Slot : DomainResource, IResource
{
    public required Base.ResourceReference Schedule { get; set; }
    public Base.CodeableConcept[]? ServiceCategory { get; set; }
    public Base.CodeableConcept[]? Specialty { get; set; }
    public required string Start { get; set; }
    public Base.CodeableConcept[]? ServiceType { get; set; }
    public Base.CodeableConcept? AppointmentType { get; set; }
    public required string Status { get; set; }
    public string? Comment { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public required string End { get; set; }
    public bool? Overbooked { get; set; }
}